packages/core/test/support.ts

```diff
@@ -278,12 +278,17 @@ export function getPoolMax(): number {
   return Config[getTestDialect()].pool?.max ?? 1;
 }
 
-type ExpectationKey = 'default' | Permutations<Dialect>;
+type ExpectationKey = 'default' | Permutations<Dialect, 4>;
 
 export type ExpectationRecord<V> = PartialRecord<ExpectationKey, V | Expectation<V> | Error>;
 
-type Permutations<T extends string, U extends string = T> =
-  T extends any ? (T | `${T} ${Permutations<Exclude<U, T>>}`) : never;
+type Decr = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
```

Comment:
I'm not sure about this type. It seems to be some sort of loop that decrements the depth with 1. But why would it continue on to 10 and isn't there a nicer way to do this? Maybe we could import a type from `type-fest` or something that might also work for larger numbers?

Reply:
That's not possible, there are no arbitrary minus operation in TypeScript types. This decrements by exploiting the positional typing of typescript's tuples

```ts
[never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][4] is the 4th element which is equal to 3 in this tuple
```